### PR TITLE
DOC: update PyArray_CheckAxis doc

### LINKED ISSUE #27191: DOC: `PyArray_CheckAxis` doesn't refer to new `NPY_RAVEL_AXIS` macro
### Issue with current documentation:

The [doc for](https://numpy.org/doc/stable/reference/c-api/array.html#c.PyArray_CheckAxis) `PyArray_CheckAxis` says that `axis=None` is checked by `*axis >= NPY_MAXDIMS`, whereas it should probably say `*axis == NPY_RAVEL_AXIS`, which is the check performed now (line 2904 of `_core/src/multiarray/ctors.c`).

### Idea or request for content:

_No response_

### PR DESCRIPTION
**Pull request by guan404ming**
## Description

This PR fix doc for PyArray_CheckAxis

close #27191

## Changes made

- change `>= NPY_MAXDIMS` to  `== NPY_RAVEL_AXIS`

### CODE REVIEW

**Comment by guan404ming** [comment]
@jorenham hi I've opened a pr, please help me review it, thanks~

**Comment by jorenham** [comment]
I don't know much about the C-API @guan404ming I'm afraid; my expertise lies with the more "high-level" stuff :)

File: doc/source/reference/c-api/array.rst [hunk 1]
```diff
@@ -688,7 +688,7 @@ From other objects
     Encapsulate the functionality of functions and methods that take
     the axis= keyword and work properly with None as the axis
     argument. The input array is ``obj``, while ``*axis`` is a
-    converted integer (so that >=MAXDIMS is the None value), and
+    converted integer (so that `*axis == NPY_RAVEL_AXIS` is the None value), and
```
Comment (ngoldbaum) [inline doc/source/reference/c-api/array.rst:691]:
```suggestion
    converted integer (so that ``*axis == NPY_RAVEL_AXIS`` is the None value), and
```

Comment (ngoldbaum) [reply doc/source/reference/c-api/array.rst:691]: Restructured text uses double backticks for code literals.

**Comment by ngoldbaum** [comment]
Thanks for the PR! Sorry for taking a little while to look it over.

**Pull request merged**